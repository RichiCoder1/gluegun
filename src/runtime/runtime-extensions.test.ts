import * as expect from 'expect'
import { Runtime } from './runtime'
import { DEFAULTS } from './run'

test('loads the core extensions in the right order', () => {
  const r = new Runtime()
  r.addCoreExtensions()
  const list = r.extensions.map(x => x.name).join(', ')

  expect(list).toBe('meta, strings, print, filesystem, semver, system, prompt, http, template, patching')
})

test('loads async extensions correctly', async () => {
  const r = new Runtime()
  r.addPlugin(`${__dirname}/../fixtures/good-plugins/async-extension`)
  r.addPlugin(`${__dirname}/../fixtures/good-plugins/threepack`)

  const toolbox = await r.run('three')
  expect(toolbox['asyncData']).toBeDefined()
  expect(toolbox['asyncData'].a).toEqual(1)
})

afterEach(() => (DEFAULTS.extensionTimeout = 10000))

test('timeouts long async extensions', async () => {
  const r = new Runtime()
  r.addPlugin(`${__dirname}/../fixtures/bad-plugins/long-async`)
  r.addPlugin(`${__dirname}/../fixtures/good-plugins/threepack`)
  DEFAULTS.extensionTimeout = 1000

  await expect(r.run('three')).rejects.toThrowErrorMatchingSnapshot()
}, 1500)
