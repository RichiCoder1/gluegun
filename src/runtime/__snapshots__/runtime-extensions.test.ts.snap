// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`timeouts long async extensions 1`] = `"The extension 'longAsyncExtension' took too long to run"`;
